couchbase-entity/src/main/java/com/kaufland/model/query/CblQueryHolder.kt

```diff
@@ -39,6 +48,16 @@ class CblQueryHolder(private val mQuery: Query) {
         return builder.build()
     }
 
+    private fun FunSpec.Builder.addQueryParamComparisonStatement(fieldHolder: CblFieldHolder, value: String) {
+        val classForTypeConversion = fieldHolder.evaluateClazzForTypeConversion()
+        addStatement(
+            "queryParams[%N] = ${TypeConversionMethodsGeneration.WRITE_METHOD_NAME}(%N, %T::class) ?: throw PersistenceException(\"Invalid type-conversion: value must not be null\")",
```

Comment:
this is the main point of the change: we use the `write()` method on the entity to run the type-conversions on the query-param.
The `findByXY...()` method does not accept null-values, therefore the type-conversion will not return null either (otherwise, something went wrong so the exception makes sense)

Reply:
`PersistenceException` is a checked exception, does this mean all query calls will need to catch this exception? Would an unchecked RuntimeException also suffice?

Reply:
Valid point. From Java-Code, yes I guess so.
From Kotlin-Code, no, because Kotlin does not care about checked exceptions.
=> but for Java-users of the framework it would be a problem, I guess.
@sbra0902 what do you think?

Reply:
I think only one java project using this library and they're not using the "query" feature so no worries for incompatiblity.
From usage perspective this type of exception is just consequent since we already throw them for the "deeper" interaction methods like "save"